### PR TITLE
Adaptor docs component

### PR DESCRIPTION
**Pull request by josephjclark**
Lots of work to do on this, just opening a PR for visibility.

Most of the time so far has been spent on setting up some new components (got my stupid hot reload working!) and refactoring describe package so that I can extend it

* [x] Create a docs component which nicely formats doc infornation
* [x] Convert describe-package tests to ava
* [ ] Refactor describe-package for a cleaner high-level API
* [ ] Add more intelligence to describe-package for richer doc information
* [x] Create a test project in examples for the new component
* [x] Send code examples to a callback (For use by clipboard & insert into monaco)
* [x] Prettify

Annoyingly I haven't been able to get Tailwind and `tsup` playing nicely together. Frustrating because I  only want one build tool in this repo. I'll have another go at some point but I also want to move forward, so I've just copied the rollup build from `workflow-diagram`.

The main point of interest is the `adaptor-docs` component. Here's my to-do list

* [x] Render signatures
* [ ] Render parameters (with good type descriptions, including complex objects)
* [ ] Render return types? Is that ever useful?
* [ ] Collapsible details for each component
* [ ] Simple Filtering (search)
* [x] Sort the list alphabetically

Most of this list will get spun out into separate PRs as we phase improvements into the initial implementation.

I don't know what to do about unit tests. We could introduce snapshot testing but I'm not keen. At the time of writing there's almost no logic to the component which even warrants testing... I

### CODE REVIEW

**Comment by josephjclark** [comment]
@stuartc This part is ready for a review - start the example with `pnpm -C examples/adaptor-docs` and go to localhost:1234.

There's plenty to do and I'll be working on this next week. This is just to get something ready for the next Lightning release.

**Comment by josephjclark** [comment]
The Lightning side of things needs testing separately but for the record, if `Lightning` is a sibling to `kit`, you can install these packages into lightning by doing:

```
$ pnpm export adaptor-docs
$ pnpm export describe-package
```

File: packages/adaptor-docs/package.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  },
+  "type": "module",
+  "module": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "author": "Open Function Group <admin@openfn.org>",
+  "license": "ISC",
+  "dependencies": {
+    "@openfn/describe-package": "workspace:^0.0.8",
+    "@openfn/runtime": "workspace:^0.0.10",
```
Comment (stuartc) [inline packages/adaptor-docs/package.json:34]: I don't believe we need this dep right?

File: packages/adaptor-docs/src/DocsPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      <h1 className="h1 text-lg font-bold text-secondary-700 mb-2">{name} v{version}</h1>
+      <div className="text-sm mb-4">Operations available for use by this adaptor are listed below.</div>
+      {functions
+        // TODO we ought to memo the sort really, although this won't render very often so it's probably ok
+        .sort((a, b) => {
+          if (a.name > b.name) return 1;
+          else if (a.name < b.name) return -1;
+          return 0;
+        })
+        .map((fn) => <Function key={fn.name} fn={fn} onInsert={onInsert} />)}
```
Comment (stuartc) [inline packages/adaptor-docs/src/DocsPanel.tsx:36]: I'm getting warnings from React here for duplicate keys, (`each`). In language-common we have `each` and `beta.each` as exports - we are going to have to add the ability to represent exported members with their namespaces.

**Review by stuartc — APPROVED** [review]
Cool I think this is a good first run at all this, it's shaping up to be pretty exciting.

I think you're aware of the edges here, but lets get this in.

I'll fixed the package import:

```
-      '@openfn/runtime': workspace:^0.0.10
```

But do have a think about namespace exported members i.e. `beta.each` and `each`.

File: packages/adaptor-docs/src/DocsPanel.tsx (hunk 2 above)
Comment (josephjclark) [reply packages/adaptor-docs/src/DocsPanel.tsx:36]: Yep, I clocked that one this morning. Let me raise an issue so it's tracked

File: packages/adaptor-docs/package.json (hunk 1 above)
Comment (josephjclark) [reply packages/adaptor-docs/package.json]: Oh good catch

Comment (stuartc) [reply packages/adaptor-docs/package.json]: I've sorted this one.

**Comment by josephjclark** [comment]
Just spun out a bunch of issues here so the remaining work (at least that identified in this thread) can now be tracked.

**Pull request merged**